ext/drx/scatter_gather_aarch64.c

```diff
@@ -623,16 +772,31 @@ drx_expand_scatter_gather(void *drcontext, instrlist_t *bb, OUT bool *expanded)
     bool res = false;
     get_scatter_gather_info(sg_instr, &sg_info);
 
-    /* Filter out instructions which are not yet supported. */
-    if (!((reg_is_z(sg_info.index_reg) || reg_is_z(sg_info.base_reg)) &&
-          sg_info.faulting_behavior == DRX_NORMAL_FAULTING)) {
-        /* We return true with *expanded=false here to indicate that no error occurred but
-         * we didn't expand any instructions. This matches the behaviour of this function
-         * for architectures with no scatter/gather expansion support.
-         */
+    /* Filter out instructions which are not yet supported.
+     * We return true with *expanded=false here to indicate that no error occurred but
+     * we didn't expand any instructions. This matches the behaviour of this function
+     * for architectures with no scatter/gather expansion support.
+     */
+    if (sg_info.faulting_behavior != DRX_NORMAL_FAULTING) {
+        /* TODO i#5036: Add support for first-fault and non-fault accesses. */
+        return true;
+    }
+    if (!reg_is_z(sg_info.base_reg) && sg_info.index_reg == DR_REG_NULL) {
+        /* TODO i#5036: Add support for scalar+immediate contiguous accesses. */
```

Comment:
scalar+imm is probably not very different from scalar+scalar right? Or is there much additional complexity? (Not suggesting that we add that support here; just curious).

Reply:
I have actually got that working too and there should be a patch for it coming soon. It's a simple change to `ext/drx/scatter_gather_aarch64.c` but it uncovered a bug in the AArch64 codec for the scalar+immediate instructions which needs to be fixed first (patch for that also coming soon).